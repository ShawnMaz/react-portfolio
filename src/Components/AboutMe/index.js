import profileImage from '../../assets/images/profile-photo.jpg';

function AboutMe(){
    return(
        <section className="container d-flex flex-column align-items-center">
            <img className='hero-img my-3' src={profileImage} alt='Shantanu Mazumder'/>
            <p className='fs-1 fw-semibold mb-0'>Shantanu Mazumder</p>
            <h6 className='fw-semibold'>Aspiring Full Stack Web Developer</h6>
            <p>Learning new skills has always been my passion. I started out my career as a Manufacturing Engineer in the Metal Fabrication industry, then moved into the Optics industry and now into Web Development. Being able to solve complex new challenges using innovative and creative solutions is what drives me to push myself to learn new skills everyday.</p>
        </section>
    );
}

export default AboutMe;