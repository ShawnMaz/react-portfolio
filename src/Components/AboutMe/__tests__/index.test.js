import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import AboutMe from "..";

// Clean memory after each test
afterEach(cleanup);

// Test
describe("About Me component", () => {
  it("renders", () => {
    render(<AboutMe />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<AboutMe />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("contains image, name, title and describtion", () => {
    render(<AboutMe />);
    expect(screen.getByTestId("profilePhoto")).toHaveAttribute(
      "src",
      "profile-photo.jpg"
    );
    expect(screen.getByTestId("profileUserName")).toHaveTextContent(
      "Shantanu Mazumder"
    );
    expect(screen.getByTestId("profileUserTitle")).toHaveTextContent(
      "Aspiring Full Stack Web Developer"
    );
    expect(screen.getByTestId("profileUserDescription")).toHaveTextContent(
      "Learning new skills has always been my passion. I started out my career as a Manufacturing Engineer in the Metal Fabrication industry, then moved into the Optics industry and now into Web Development. Being able to solve complex new challenges using innovative and creative solutions is what drives me to push myself to learn new skills everyday."
    );
  });
});
