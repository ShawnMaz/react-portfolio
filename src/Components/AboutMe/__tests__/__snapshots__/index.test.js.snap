// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Me component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container d-flex flex-column align-items-center"
  >
    <h2
      class="my-3"
    >
      About Me
    </h2>
    <img
      alt="Shantanu Mazumder"
      class="hero-img my-3"
      data-testid="profilePhoto"
      src="profile-photo.jpg"
    />
    <p
      class="fs-1 fw-semibold mb-0"
      data-testid="profileUserName"
    >
      Shantanu Mazumder
    </p>
    <h6
      class="fw-semibold"
      data-testid="profileUserTitle"
    >
      Aspiring Full Stack Web Developer
    </h6>
    <p
      data-testid="profileUserDescription"
    >
      Learning new skills has always been my passion. I started out my career as a Manufacturing Engineer in the Metal Fabrication industry, then moved into the Optics industry and now into Web Development. Being able to solve complex new challenges using innovative and creative solutions is what drives me to push myself to learn new skills everyday.
    </p>
  </section>
</DocumentFragment>
`;
